content/en/docs/Fundamentals/Services/dependency-injection.md

```diff
@@ -5,12 +5,154 @@ linkTitle: "Dependency Injection"
+
+```C#
+using DependencyInjectionSample.Interfaces;
+using DependencyInjectionSample.Services;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+
+builder.Services.AddScoped<IMyDependency, MyDependency>();
+
+var app = builder.Build();
+```
+##What are Service lifetimes?
+
+There are three types of service lifetimes
+1. Scoped
+2. Transient
+3. Singleton
+
+##Scoped
+
+Service scope is a type of method that allows two different page managers to share the same dependencies.
+
+Usually, this is not possible, but service scopes allow this by creating an overhanging method 
+Another unique aspect of service scopes is that it allows for a ServiceKey to provide a default implementation.
+
+For web applications, scoped lifetimes are created once per user request.
+
+You usually register scoped services with AddScoped
+###Service Scope Constructors
+**Below constructs a new scope for a service scope class.**
+protected constructor(parent: ServiceScope | undefined);
+
+[Service Scope Method Details](https://docs.microsoft.com/en-us/javascript/api/sp-core-library/servicescope?view=sp-typescript-latest#:~:text=ServiceScope%20provides%20a%20formalized%20way,dependencies%20in%20an%20extensible%20way.)
+[More Method Details Link](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addscoped?view=dotnet-plat-ext-6.0)
+
+##Transient
+
+Transient lifetime services are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services. Register transient services with AddTransient
+
+In apps that process requests, transient services are disposed of at the end of the request.
+
+[Transient lifetime methods](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions.addtransient?view=dotnet-plat-ext-6.0)
+
+##Singleton
+
+These are a type of services that usually are created either the first time they are requested or when providing an implementation instance directly to the container.
+
+Every subsequent request of the service implementation from the dependency injection container uses the same instance. If the app requires singleton behavior, allow the service container to manage the service's lifetime. Don't implement the singleton design pattern and provide code to dispose of the singleton. Services should never be disposed of by code that resolved the service from the container. If a type of factory is registered as a singleton, the container disposes of the singleton automatically.
+
+##How do you make a Service?
+![Tutorial](https://csharpcorner-mindcrackerinc.netdna-ssl.com/article/create-windows-services-in-c-sharp/Images/image002.png)
+
+*In Visual Studios,  go to File > New > Project and select �Window Service�*
+
+*Right-click screen > Add installer*
+
+*Right-click screen > View code (F7)*
+
+*Type out your code*
```

Comment:
Rewrite this sentence with something more professional, such as "Start programming your service"